client-ts/rollup-base.js

```diff
@@ -0,0 +1,55 @@
+import path from 'path';
+
+import resolve from 'rollup-plugin-node-resolve'
+import sourceMaps from 'rollup-plugin-sourcemaps'
+import commonjs from 'rollup-plugin-commonjs'
+
+let polyfills = [ 'es6-promise', 'buffer', 'base64-js', 'ieee754' ];
+let allowed_externals = [ ];
+
+export default function(rootDir, moduleGlobals) {
+    let pkg = require(path.join(rootDir, "package.json"));
+    return {
+        input: path.join(rootDir, "dist", "cjs", "browser-index.js"),
+        output: {
+            file: pkg.browser,
+            format: "umd",
+            name: pkg.umd_name,
+            sourcemap: true,
+            banner: "/* @license\r\n" +
+                " * Copyright (c) .NET Foundation. All rights reserved.\r\n" +
+                " * Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.\r\n" +
+                "*/",
+            globals: moduleGlobals,
+        },
+        external: function (m) {
```

Comment:
This function ensures that things in `node_modules` aren't "rolled up" into the browser file. **EXCEPT** for the polyfills we explicitly allow, listed above.

Reply:
More context: This function is called with the right-side of every `import {} from "foo"`, and given the **resolved** path for `"foo"`. So for things within our own source, we return `false`, indicating it is NOT external and should be rolled-up. For things in `node_modules`, we check against the polyfill list.